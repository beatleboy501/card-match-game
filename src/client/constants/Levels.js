export default Object.freeze({
  EASY: 'easy',
  HARD: 'hard',
});
