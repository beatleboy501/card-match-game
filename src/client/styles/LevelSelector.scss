.buttonSet {
  margin-top: 2em;
  margin-bottom: 2em;
}

.button {
  height: 100px;
  width: 100px;
  margin-top: 1em;
  margin-bottom: 1em;
}

.heading {
  display: flex !important;
  justify-content: center;
  align-items: center;
  height: 3em;
}
