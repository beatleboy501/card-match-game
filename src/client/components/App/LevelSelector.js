import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import styles from '../../styles/LevelSelector.scss';

const propTypes = {
  fetchCards: PropTypes.func.isRequired,
  levels: PropTypes.arrayOf(PropTypes.object),
};

const defaultProps = {
  levels: [],
};

const LevelSelector = (props) => {
  const { fetchCards, levels } = props;
  return (
    <div className="level-selector">
      {levels.length ? (
        <Typography className={styles.heading} variant="title" color="inherit">
          Choose a Level
        </Typography>
      ) : <div />}
      <div className={styles.buttonSet}>
        {levels.map(level => (
          <Button
            key={level.difficulty}
            className={styles.button}
            onClick={() => fetchCards(level.difficulty)}
            type="button"
          >
            {level.difficulty}
          </Button>
        ))}
      </div>
    </div>
  );
};

LevelSelector.propTypes = propTypes;
LevelSelector.defaultProps = defaultProps;

export default LevelSelector;
