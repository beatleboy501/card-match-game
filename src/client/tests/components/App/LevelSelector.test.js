import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LevelSelector from '../../../components/App/LevelSelector';
import Levels from '../../../constants/Levels';

Enzyme.configure({ adapter: new Adapter() });

const levels = [
  { difficulty: Levels.EASY, cards: ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h'] },
  { difficulty: Levels.HARD, cards: ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p'] },
];

const fetchCards = (level) => {
  const { cards } = levels.length ? levels.find(l => l.difficulty === level) : { cards: [] };
  return cards;
};

let component;

it('Renders without crashing', () => {
  component = shallow(<LevelSelector fetchCards={fetchCards} levels={levels} />).shallow();
  expect(component).toMatchSnapshot();
});
