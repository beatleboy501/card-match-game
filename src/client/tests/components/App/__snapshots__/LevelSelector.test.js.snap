// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="level-selector"
  >
    <WithStyles(Typography)
      className="heading"
      color="inherit"
      variant="title"
    >
      Choose a Level
    </WithStyles(Typography)>
    <div
      className="buttonSet"
    >
      <WithStyles(Button)
        className="button"
        onClick={[Function]}
        type="button"
      >
        easy
      </WithStyles(Button)>
      <WithStyles(Button)
        className="button"
        onClick={[Function]}
        type="button"
      >
        hard
      </WithStyles(Button)>
    </div>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          className="heading"
          color="inherit"
          variant="title"
        >
          Choose a Level
        </WithStyles(Typography)>,
        <div
          className="buttonSet"
        >
          <WithStyles(Button)
            className="button"
            onClick={[Function]}
            type="button"
          >
            easy
          </WithStyles(Button)>
          <WithStyles(Button)
            className="button"
            onClick={[Function]}
            type="button"
          >
            hard
          </WithStyles(Button)>
        </div>,
      ],
      "className": "level-selector",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Choose a Level",
          "className": "heading",
          "color": "inherit",
          "variant": "title",
        },
        "ref": null,
        "rendered": "Choose a Level",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              className="button"
              onClick={[Function]}
              type="button"
            >
              easy
            </WithStyles(Button)>,
            <WithStyles(Button)
              className="button"
              onClick={[Function]}
              type="button"
            >
              hard
            </WithStyles(Button)>,
          ],
          "className": "buttonSet",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "easy",
            "nodeType": "class",
            "props": Object {
              "children": "easy",
              "className": "button",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "easy",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "hard",
            "nodeType": "class",
            "props": Object {
              "children": "hard",
              "className": "button",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "hard",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            className="heading"
            color="inherit"
            variant="title"
          >
            Choose a Level
          </WithStyles(Typography)>,
          <div
            className="buttonSet"
          >
            <WithStyles(Button)
              className="button"
              onClick={[Function]}
              type="button"
            >
              easy
            </WithStyles(Button)>
            <WithStyles(Button)
              className="button"
              onClick={[Function]}
              type="button"
            >
              hard
            </WithStyles(Button)>
          </div>,
        ],
        "className": "level-selector",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Choose a Level",
            "className": "heading",
            "color": "inherit",
            "variant": "title",
          },
          "ref": null,
          "rendered": "Choose a Level",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                className="button"
                onClick={[Function]}
                type="button"
              >
                easy
              </WithStyles(Button)>,
              <WithStyles(Button)
                className="button"
                onClick={[Function]}
                type="button"
              >
                hard
              </WithStyles(Button)>,
            ],
            "className": "buttonSet",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "easy",
              "nodeType": "class",
              "props": Object {
                "children": "easy",
                "className": "button",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "easy",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "hard",
              "nodeType": "class",
              "props": Object {
                "children": "hard",
                "className": "button",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "hard",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
